src/traces/pie/index.js

```diff
@@ -373,6 +373,12 @@ pie.plot = function(gd, cdpie) {
                     }
                 }
 
+                function handleClick (evt) {
+                    gd._hoverdata = pt;
+                    gd._hoverdata.trace = slices[0][0].__data__.trace;
```

Comment:
Right. My mistake.

Try `cd[0].trace` like in https://github.com/plotly/plotly.js/blob/master/src/plots/cartesian/graph_interact.js#L437

to grab a ref to the trace.